packages/data-broker/src/services/Service.ts

```diff
@@ -62,11 +60,20 @@ export abstract class Service {
     options: DeleteOptions,
   ): Promise<Diagnostics>;
 
-  public abstract pull(
-    dataSources: DataSource[],
-    type: DataSourceType,
-    options: PullOptions,
-  ): Promise<AnalyticResults | EventResults | SyncGroupResults> | never;
+  public abstract pullAnalytics(
+    dataElements: DataElement[],
+    options: Record<string, unknown>,
```

Comment:
I simplified the options type to `Record<string, unknown>`. This is because we shouldn't leak any abstraction details unless we have to, and there is no such need here.